### PR TITLE
Fixes for logger getting cleaned up too early, also handles null term…

### PR DESCRIPTION
**Pull request by JonathanHenson**
…inated buffers properly.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: source/io.c [hunk 1]
```diff
@@ -259,6 +202,16 @@ PyObject *aws_py_io_client_tls_ctx_new(PyObject *self, PyObject *args) {
     if (certificate_buffer && private_key_buffer && certificate_buffer_len > 0 && private_key_buffer_len > 0) {
         struct aws_byte_cursor cert = aws_byte_cursor_from_array(certificate_buffer, certificate_buffer_len);
         struct aws_byte_cursor key = aws_byte_cursor_from_array(private_key_buffer, private_key_buffer_len);
+        
+       /* adding the null terminator in python land screws up the byte buf apis. We never consider the null terminator
```
Comment (justinboswell) [inline source/io.c]: Ah, fuck, that's my fault. How did this ever work ever?

File: source/module.c [hunk 2]
```diff
@@ -29,6 +29,53 @@
...
+        .level = log_level,
+        .file = NULL,
+        .filename = NULL,
+    };
+
+    Py_ssize_t stdout_len = (Py_ssize_t)strlen("stdout");
+
+    Py_ssize_t cmp_len = file_path_len > stdout_len ? stdout_len : file_path_len;
+
+    if (!memcmp("stdout", file_path, (size_t)cmp_len)) {
```
Comment (justinboswell) [inline source/module.c]: Should we allow these to be case-insensitive?

Comment (JonathanHenson) [reply source/module.c]: I don't think the extra branches are worth it, and it's written in lowercase in python anyways.

File: source/io.c (hunk 1 above)
Comment (JonathanHenson) [reply source/io.c]: it worked fine till we started loading certs on apple, and it did not like that NULL at all

**Pull request merged**